prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -91,48 +93,55 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
         // get the file name, or create one from the report ID, NY requires a file name in the POST
         val fileName = header.reportFile.externalName ?: "$reportId.hl7"
         // get the username/password to authenticate with OAuth
-        val credential: RestCredential = lookupDefaultCredential(receiver)
+        val credential: RestCredential = if (restTransportInfo.authType != "two-legged")
+            lookupDefaultCredential(receiver)
```

Comment:
I think it's better if lookupDefaultCredential goes in the else and the if  checks for authType == two-legged

Reply:
Will do.  Thanks